### PR TITLE
squid: add version 6.7 to extra.omnios

### PR DESCRIPTION
**Pull request by cgrzemba**
As announced at FOSSDEM, I am sending my work regarding quid here.

### CODE REVIEW

**Comment by citrus-it** [comment]
Thanks @cgrzemba, we'll take a look!

File: build/squid/build.sh [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+. ../../lib/build.sh
+
+PROG=squid
+VER=6.7
+PKG=ooce/network/proxy/squid
+SUMMARY="Squid WEB Proxy"
+DESC="Squid is a caching proxy for the Web supporting HTTP, HTTPS, FTP, and more."
+
+set_mirror "http://www.squid-cache.org/Versions/v6/"
+set_checksum sha256 "e14daa4eae41925d1ae3f08e64439a6aaa3011bdced686628b8de697d5ab8428"
```
Comment (hadfl) [inline build/squid/build.sh:26]: I've put the source on the OmniOS mirror. You can drop these two lines

File: build/squid/build.sh [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    --libdir=$PREFIX/lib/$ISAPART64
+"
+
+CXXFLAGS+=" -Wno-unknown-pragmas -Wno-deprecated-declarations"
+export LIBLDAP_LIBS="-lldap -llber"
+export LIBLDAP_PATH="-L$OPREFIX/lib/amd64 "
+LDFLAGS[amd64]+=" -Wl,-z -Wl,ignore -L$OPREFIX/lib/amd64 -R$OPREFIX/lib/amd64"
+
+init
+download_source "" $PROG $VER
```
Comment (hadfl) [inline build/squid/build.sh:119]: Using the OmniOS mirror, you'll have to change this to `download_source $PROG $PROG $VER`

File: build/squid/files/squid.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        <documentation>
+          <doc_link name='squid-cache.org' uri='http://www.squid-cache.org'/>
+          <manpage title='squid' section='8' manpath='/usr/share/man'/>
+        </documentation>
+      </template>
+    </instance>
+    <stability value='Evolving'/>
+  </service>
+</service_bundle>
+
```
Comment (hadfl) [inline build/squid/files/squid.xml:34]: can please you run `./tools/tidy-manifests` on the manifest file so it'll be formatted consistently with the other manifests

File: build/squid/local.mog [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#
+# Copyright 2020 Carsten Grzemba
+
+license COPYING license=GPLv2
+
+<include binlink.mog>
+<include manlink.mog>
+
+group groupname=$(GROUP) gid=84
+user ftpuser=false username=$(USER) uid=84 group=$(GROUP) \
```
Comment (hadfl) [inline build/squid/local.mog:19]: gid 84 collides with `nagcmd`. Please have a look at `doc/idlist.md` and select a free uid/gid. Then please document it in `doc/idlist.md`

File: build/squid/local.mog [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+<transform dir path=var/opt/ooce/squid/run(/.*)* \
+	-> drop >
+
+<transform dir path=var/$(PREFIX)(/.*)* \
+	-> default group squid >
+<transform dir path=var/$(PREFIX)(/.*)* \
+	-> default owner squid>
+<transform dir path=var/run \
+	-> drop >
```
Comment (hadfl) [inline build/squid/local.mog:39]: please change tabs to 4 spaces for consistent formatting

File: build/squid/build.sh [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    --enable-wccp
+    --enable-wccpv2
+    --disable-arch-native
+    --disable-esi
+    ac_cv_path_krb5_config=/usr/bin/krb5-config
+    squid_cv_OpenLDAP=0
+"
+
+CONFIGURE_OPTS_64+="
+    --libdir=$PREFIX/lib/$ISAPART64
```
Comment (hadfl) [inline build/squid/build.sh:110]: `CONFIGURE_OPTS_64` and `$ISAPART64` do no longer exist. You'll have to change it to `CONFIGURE_OPTS[amd64]+=" $OPREFIX/lib/amd64"`

**Review by hadfl — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Pull request merged**